### ISSUE TITLE
Nvidia Shield 4k HDR Frame Rate Playback issue

### ISSUE THREAD

**Issue by sendxmedic** [koying/SPMC#1163]
Hello,

When i change the setting to Allow hardware acceleration - MediaCodec (Surface) to enable 4K HDR it seems to auto play any move i choose to play at 60hz vs when the codec is off and it looks like i get the proper frame rate. This seems to be the case regardless of whether i use Kodi or SPMC. Is there a fix for this auto switching framerate issue or is this something that i will have to live with if i want HDR?

Edit: i have a 55" LG Oled and when i trigger the HDR prompt at the top i check the HDR settings at all true motion is set to off but it still plays the 4K videos at the 60hz vs the 24hz movies regularly play at.

**Comment by chrisc0le** [comment]
Hi sendxmedic

I also have an LG oled tv (B6V). Can you tell me how you know the refresh rate is 60Hz?
There is no INFO button on my remote and the only info I can get is by clicking on the input banner on the top left which displays the resolution and colour space etc. But not refresh rate.

The shield says it's outputting at 23.976 (click the right joystick on the controller), and my AVR confirms it's receiving 23.976 and also outputting 23.976 to the TV.

But what the TV is displaying is a mystery.

This is with both Media codec's enabled and the media is HDR (confirmed by the HDR banner top right).

**Comment by sendxmedic** [comment]
Its easily noticeable that your are getting the Soap Opera Effect i cant be too certain that its 60hz but its definitely not switching framerates correctly to match the Lg B6 oled. If you turn off the Mediacodec (surface) you will see that the video is running at a much lower frame rate and lesser quality due to no HDR. Ive tried different settings turning them on and off but nothing seems to alleviate the Soap Opera Effect when trying to watch movies in HDR.

**Comment by chrisc0le** [comment]
That is weird. Mine plays Ok, but I can spot frame duplication every now and then (probably due to 24Hz on the Tv and 23.976 fps). With Sync to display on it's smoother but the audio is re sampled then and doesn't sound nearly as good as passthrough. I've come to the conclusion that nothing will resolve this.

It sounds to me like a true motion effect. Do you have 'real cinema' enabled? The default HDR picture modes have true motion enabled. I went in and turned everything except 'real cinema' off and adjusted the picture too.

**Comment by sendxmedic** [comment]
Yeah I checked my TVs settings and thought that might have been the reason but even disabling everything in true motion same result. It’s a brain buster. I may just have to return my nvidia shield tv because I just can’t stand the effect and jutter in fast moving scenes makes it unwatchable.

**Comment by chrisc0le** [comment]
Just a thought. I fell into this trap when the Shield FW updated recently.
There are some colour spaces that will only allow '3' refresh rates on the TV.

You may have one of those set.
In the shield (android) display settings/Advanced Settings/HDMI settings.
I had to use YUV 422 12-Bit Rec. 2020. because all the others (other than RGB Rec.709) only allowed 3 refresh rate options. 

Give it a check.

**Comment by sendxmedic** [comment]
I’ll definitely give a go when I get off work. Thanks for info!

**Comment by sendxmedic** [comment]
I tried these settings and it doesnt seem to really help with the Frame rate issue. The odd thing is i checked the Vudu app on my Nvidia Shield and my movies through there do the same thing. But when i go over to my Apple Tv 4k it is completely fine and seems to output the correct frame rates. It definitely seems to be an issue with the Nvidia Shield which is unfortunate because its a great device.
